test/tests/functional/pbs_power_provisioning_sgi.py

```diff
@@ -60,8 +93,23 @@ def setUp(self):
             host = n['Mom']
             if host is None:
                 continue
-            if host == self.server.hostname:
+            if host == self.server.shortname:
                 self.server.manager(MGR_CMD_DELETE, NODE, None, host)
+        # setup environment for power provisioning
+        self.power_nodes = self.setup_sgi_api(self.script)
+        if(self.power_nodes == 0):
+            self.skip_test("No mom found with power profile setup")
+        else:
+            if(self.check_mom_configuration()):
+                # enable power hook
+                self.enable_power()
+                power_moms = self.moms
```

Comment:
You can directly use self.moms and iterate through the dict.

Reply:
@visheshh  do we need to iterate all moms here or only those moms where SGI script has been set successfully ?

Reply:
Yes